### ISSUE TITLE
[1.5.0-RC1] Eviction error fails fast with `dependsOn`

### ISSUE THREAD

**Issue by kubukoz** [sbt/sbt#6399]
## steps

1. Define a build with multiple modules, one of which depends on the other one, both of which have dependencies that contain a conflict (as per the new eviction schemes):

```scala
ThisBuild / libraryDependencySchemes += "io.circe" %% "circe-core" % "early-semver"

val core = project.settings(
  libraryDependencies ++= Seq(
    "co.fs2" %% "fs2-core" % "2.5.3",
    // conflicts with transitive version pulled in by fs2 (2.3.x)
    "org.typelevel" %% "cats-effect" % "3.0.0-RC2"
  )
)

val main = project
  .settings(
    libraryDependencies ++= Seq(
      "io.circe" %% "circe-generic-extras" % "0.13.0",
      // conflicts with transitive veresion pulled in by circe-generic-extras (0.13.x)
      "io.circe" %% "circe-core" % "0.14.0-M4"
    )
  )
  .dependsOn(core)

val root = project.in(file(".")).aggregate(core, main)
```

2. Cause library version conflicts in `core` *and* `main`

3. Run `update`

sbt: 1.5.0-RC1

## problem

Only errors from `core` show up:

```
sbt:root> update
[error] stack trace is suppressed; run last core / update for the full output
[error] (core / update) found version conflict(s) in library dependencies; some are suspected to be binary incompatible:
[error]
[error] 	* org.typelevel:cats-effect_2.12:3.0.0-RC2 (early-semver) is selected over 2.3.3
[error] 	    +- core:core_2.12:0.1.0-SNAPSHOT                      (depends on 3.0.0-RC2)
[error] 	    +- co.fs2:fs2-core_2.12:2.5.3                         (depends on 2.3.3)
[error]
[error]
[error] this can be overridden using libraryDependencySchemes or evictionErrorLevel
[error] Total time: 0 s, completed 15 Mar 2021, 21:40:56
```

## expectation

Errors from `main` also appear. This happens if there's no `dependsOn` (presumably because `root` only aggregates the projects and that has a different effect on eagerly failing the `update` task):

```
[error] stack trace is suppressed; run last core / update for the full output
[error] stack trace is suppressed; run last main / update for the full output
[error] (core / update) found version conflict(s) in library dependencies; some are suspected to be binary incompatible:
[error]
[error] 	* org.typelevel:cats-effect_2.12:3.0.0-RC2 (early-semver) is selected over 2.3.3
[error] 	    +- core:core_2.12:0.1.0-SNAPSHOT                      (depends on 3.0.0-RC2)
[error] 	    +- co.fs2:fs2-core_2.12:2.5.3                         (depends on 2.3.3)
[error]
[error]
[error] this can be overridden using libraryDependencySchemes or evictionErrorLevel
[error] (main / update) found version conflict(s) in library dependencies; some are suspected to be binary incompatible:
[error]
[error] 	* io.circe:circe-core_2.12:0.14.0-M4 (early-semver) is selected over 0.13.0
[error] 	    +- main:main_2.12:0.1.0-SNAPSHOT                      (depends on 0.14.0-M4)
[error] 	    +- io.circe:circe-generic_2.12:0.13.0                 (depends on 0.13.0)
[error]
[error]
[error] this can be overridden using libraryDependencySchemes or evictionErrorLevel
[error] Total time: 0 s, completed 15 Mar 2021, 21:41:19
```

## notes

Maybe this is intentional, I just think it would be a more pleasant UX if you could see all the issues at once.

Here's a reproduction: https://github.com/kubukoz/sbt-conflict-demo

**Comment by eed3si9n** [comment]
`main/update` task depends on the `core/update`. This is a synthetic connection as far as I can tell, but at some point in the history it was requested as a feature https://github.com/sbt/sbt/issues/246, which says that an invalidation of `core/update` must invalidate `main/update`. I think this matters because `update` is not always an immutable, repeatable function. The content of `-SNAPSHOT` can change, and you'd want the subprojects to be consistent when it happens?

Either case, I don't think I'd categorize this to the severity of "bug."

**Comment by eed3si9n** [comment]
From the performance aspect, however, it might be worth investigating if we can parallelize the `update` task of dependent subprojects.
If `update` takes 10s, and you have 3-level deep subprojects, blocking on the `dependsOn` would mean waiting 20s just to get to the `update`, which is a behavior I've noticed in super shell.

**Comment by kubukoz** [comment]
> Either case, I don't think I'd categorize this to the severity of "bug."

It's an inconvenience (so probably just a lack of a feature). One might think they've solved all the conflicting dependencies because they fixed the last one in a subproject, only to realise another one was depending on that project and its `update` never ran - so next time they run `update` they'll see a bunch of more errors.

So I guess if the parallelization of `update` was possible, we would get both the possible performance gains and more visibility too :)

**Comment by eed3si9n** [comment]
> So I guess if the parallelization of update was possible, we would get both the possible performance gains and more visibility too :)

But do you see that making it concurrent (as in semantically independent) would undo the fix for #246?
This could mean that when `core/update` is invalidated for whatever reason, that will no longer trigger `update` invalidation of downstream subprojects. If the `update` is already started, you can't get invalidated. This could result to potentially weird situation where `core` is compiled against Guava 21, but the rest are compiled against Guava 20 due to dynamic version resolution, which then could trigger runtime errors.

**Comment by kubukoz** [comment]
Yeah, I see it now. Thanks for the explanation :)
